src/tests/compile.js

```diff
@@ -1,68 +1,114 @@
+  "disallow_empty_diverts.ink",
+  "divert_not_found_error.ink",
+  "argument_name_collisions.ink",
+  "function_call_restrictions.ink",
+  "function_purity_checks.ink",
+  "wrong_variable_divert_target_reference.ink",
+  "empty_thread_error.ink",
+  "const_redefinition.ink",
+  "require_variable_targets_typed.ink",
+  "temp_not_allowed_cross_stitch.ink",
+  "nested_choice_error.ink",
+  "stitch_naming_collision.ink",
+  "end_of_content_function.ink",
+  "end_of_content_return_statement.ink",
+  "variable_naming_collision_with_arg.ink",
+  "variable_naming_collision_with_flow.ink",
+  "variable_name_colision_with_flow.ink",
+  "variable_name_collision_with_arg.ink",
+  "weave_point_naming_collision.ink",
+];
+
+function runInklecate(input, output, inklecate, extraArgs) {
+  let command = `${inklecate} ${extraArgs} -o "${output}" "${input}"`;
+
+  return new Promise((resolve, reject) => {
+    childProcess.exec(command, (error, stdout) => {
+      if (error) {
+        let fileName = path.basename(input);
+        if (filesExpectedToFailCompilation.includes(fileName)) {
+          resolve(stdout);
+          return;
+        }
+
+        // Adding stdout as well, in case this is a compilation error.
+        reject(new Error(`${error.message.replace(/\n+$/, "")}\n${stdout}`));
+      } else {
+        resolve(stdout);
+      }
+    });
+  });
 }
 
-async function compileInkFile() {
-	console.log("Compiling test cases…");
+async function compileInkFile(inklecate) {
+  console.log("Compiling test cases…");
+
+  let files = glob.sync(`${inkFileDirectory}/**/!(includes)/*.ink`);
 
-	let files = glob.sync(`${inkFileDirectory}/**/!(includes)/*.ink`);
+  let promises = files.map(async (file) => {
+    let out = path.join(
+      compiledFileDirectory,
+      path.relative(inkFileDirectory, file) + ".json"
+    );
+    let fileName = path.basename(file);
 
-	let promises = files.map(async (file) => {
-		let out = path.join(compiledFileDirectory, path.relative(inkFileDirectory, file) + '.json');
-		let fileName = path.basename(file)
+    let extraArgs = "";
 
-		let extraArgs = '';
+    if (filesRequiringCFlag.includes(fileName)) {
+      extraArgs = "-c";
+    }
 
-		if (filesRequiringCFlag.includes(fileName)) {
-			extraArgs = '-c';
-		}
+    let outDirectory = path.dirname(out);
 
-		let outDirectory = path.dirname(out);
+    await fs.ensureDir(outDirectory);
+    return runInklecate(file, out, inklecate, extraArgs);
+  });
 
-		await fs.ensureDir(outDirectory);
-		return runInklecate(file, out, extraArgs);
-	})
+  let results = await Promise.allSettled(promises);
+  let errors = results.filter((result) => result.status === "rejected");
+
+  if (errors.length === 0) {
+    console.log("Done.");
+  } else {
+    errors.forEach((error) => console.error(`\n${error.reason.message}`));
+  }
+}
 
-	let results = await Promise.allSettled(promises);
-	let errors = results.filter(result => result.status === "rejected");
+let inklecate = process.argv[3];
```

Comment:
Since the tests are autocompiling now, this file is back to using the original inklecate. An improvement: it can now be provided, instead of relying on its presence in $PATH. We'll use the json stored in the repo to perform a diff, but that'll come in a future PR.